### DISCUSSION TITLE
Example code for using OIDC authentication?

### DISCUSSION THREAD

**Discussion by mfinnigan** [rajanadar/VaultSharp#349] [Q&A]
Using the vault binary, it's super easy to get a human user OIDC authed to Vault. They declare their role name, the vault client bounces up a web browser to get or check their auth status against the idP, and away you go.

The VaultSharp library, by contrast, is expecting a role name and a JWT. I don't see any example code on how one might generate a JWT (when you're authing with OIDC specifically). I do see the Non Login methods in [IJWTAuthMethod.cs](https://github.com/rajanadar/VaultSharp/commit/8478677d1b60d902d1a30016035ac5a2fe637f18#diff-21e6b73eb2be7890c4d05f9a040478df2e78d3b95ad4829b2645f5683b8c736f) 

but I'm unclear on how to use them, or if the "OIDC ID Token" returned by DoOIDCCallbackAsync is the same thing as a JWT token in the first place.

I don't see anything at all on github that shows someone using these methods in their own code, so I'm having a tough time answering this on my own.

**Comment by rajanadar** [comment]
hi @mfinnigan 

Can you please provide details of the vault cli commands and api you're talking about?
